sdk/storage/tests/data_lake.rs

```diff
@@ -29,13 +29,17 @@ async fn test_data_lake_file_system_functions() -> Result<(), Box<dyn Error + Se
     let resource_id = "https://storage.azure.com/";
     let bearer_token = DefaultCredential::default().get_token(resource_id).await?;
 
-    let data_lake_client = storage_account_client
-        .as_storage_client()
-        // This test won't work during replay in CI until all operations are converted to pipeline architecture
-        // .as_data_lake_client_with_transaction(account, bearer_token.token.secret().to_owned(), "test_data_lake_file_system_functions")?;
-        .as_data_lake_client(account, bearer_token.token.secret().to_owned())?;
+    let storage_client = storage_account_client.as_storage_client();
+    let data_lake_client = DataLakeClient::new(
```

Comment:
Have to call `.new()` now instead of `as_...()`. Couldn't make the mismatch between Arc<SomeClient> and SomeClient work any other way.